visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -1,15 +1,96 @@
 import { CodeMapNode } from "../codeCharta.model"
+import { ValidationError, Validator, ValidatorResult } from "jsonschema"
+
+const jsonSchema = require("./schema.json")
+const latestApiVersion = require("../../../package.json").codecharta.apiVersion
+
+interface ApiVersion {
+	major: number
+	minor: number
+}
+
+export interface CCValidationResult {
+	error: string[]
+	warning: string[]
+	title: string
+}
 
 export class FileValidator {
-	private static hasUniqueChildren(node: CodeMapNode) {
+	private static ERROR_MESSAGES = {
+		fileIsInvalid: {
+			title: "Error Loading File",
+			message: "file is empty or invalid"
+		},
+		apiVersionIsInvalid: {
+			title: "Error File API Version",
```

Comment:
Lets also remove the word `File` here, so we call the `API Version` always the same like below (:

Reply:
But the other API Version Errors are related to the API Version of CodeCharta, e.g. when using the App 
Might be confusing?

Reply:
I guess we could remove it from the title and leave it in the text?

Reply:
Ah shit. Now I get it. I got confused as well. You are right

Reply:
mhm no I don't get it. Aren't we only talking about one API Version? which is the `1.1` at the moment?

Reply:
Minor/ Major Api Version wrong refer to CodeCharta
Richard asked for this feature if the CodeCharta Version is behind the version for the file we throw an error/a warning
apiVersionIsInvalid refers to an invalid file api, e.g. 1.a

Reply:
Well its the CodeCharta project. Everything refers to CodeCharta I guess.. Its always the "CodeCharta `cc.json` API Version" no matter if its valid, outdated or invalid

Reply:
ok, the first two refer to the app version
It is specifically asked for in the issue, due to a possible 2.0 cc.json with fundamental changes

Reply:
The app version is the latestApiVersion is guess (which we store in the package.json)?

Reply:
yes, but there is an app version of codecharta as well, right?

Reply:
eh no.. we have a codecharta version which is like `1.45.5` the one Ben changes when he does the releases every few weeks, and there is the cc.json API version, which is `1.1` since almost forever

Reply:
The issue asked to check whether the api Version in the  cc.json is lower than the api Version of the loaded file
If sich a case is impossible, maybe we should change the request accordingly?

Reply:
You sentence is misunderstanding, but I think I get what you mean.

That can happen when:
- somebody installs codecharta locally with `nom install codecharta-visualization` with CC Version 1.45.5 and a latest cc.json API Version of 1.1
- one year goes by
- we (the CC Team) updated the API Version to 2.0 Kollege
- the user imports a file (from a colleague) with API Version 1.1 cc.json into the old CodeCharta application
- now the loaded file has a newer version than the CC Application wants

Reply:
Ok, then I fail to see a reason to change the message

This message is thrown if the api Version of the loaded fail is not valid 

The other two are thrown in one of the case described above and are related to the cc.json only

Reply:
I can change it, but I believe it would result in more confusion

Reply:
It's just about the usage of the word "file" twice here. We use it when the file context is actually invalid in the first part, but when we speak about the API Version, we also use the term `File API Version` even though the API Version is obviously part of the file.
The suggestion is to remove the word "file" from the API Version checks. I agree with that.

Reply:
Ok, I think I understand what you mean now